spec/models/salt_event_spec.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+require "rails_helper"
+
+describe SaltEvent do
+  describe ".process_next_event" do
+    it "processes new-minion events" do
+      event_data = "{\"_stamp\": \"2017-01-24T13:30:20.794326\", "\
+        "\"pretag\": null, \"cmd\": \"_minion_event\", \"tag\": \"minion_start\", "\
+        "\"data\": \"Minion MyMinion started at Tue Jan 24 13:30:20 2017\", "\
+        "\"id\": \"MyMinion\"}"
+
+      event = FactoryGirl.create(
+        :salt_event, tag: "minion_start", data: event_data)
+
+      expect{
+        SaltEvent.process_next_event(worker_id: "MyWorker")
+      }.to change{ Minion.where(hostname: "MyMinion").count }.by(1)
+    end
+
+    it "processes irrelevant events" do
+      event = FactoryGirl.create(:salt_event, tag: "nobody_cares_about_this_event")
+
+      expect{
+        SaltEvent.process_next_event(worker_id: "MyWorker")
+      }.to change{ SaltEvent.where(processed_at: nil).count }.from(1).to(0)
+    end
+
+    it "processes events not assigned to any worker" do
+      event = FactoryGirl.create(:salt_event, worker_id: nil)
+      expect {
+        SaltEvent.process_next_event(worker_id: "MyWorker")
+      }.to change{ event.reload.processed_at }.from(nil).to(Time)
+    end
+
+    it "processes events assigned to the same worker (leftovers from dying?)" do
+      event = FactoryGirl.create(
+        :salt_event, worker_id: "this_worker", processed_at: nil,
+        taken_at: Time.current - (SaltEvent::PROCESS_TIMEOUT_MIN - 2).minutes) # not timed out
+
+      expect {
+        SaltEvent.process_next_event(worker_id: "this_worker")
+      }.to change{ event.reload.processed_at }.from(nil).to(Time)
+    end
+
+    it "processes events assigned to the any worker but not completed for more than PROCESS_TIMEOUT_MIN minutes" do
+      event = FactoryGirl.create(
+        :salt_event, worker_id: "other_worker", processed_at: nil,
+        taken_at: Time.current - (SaltEvent::PROCESS_TIMEOUT_MIN + 2).minutes)
+
+      expect {
+        SaltEvent.process_next_event(worker_id: "this_worker")
+      }.to change{ event.reload.processed_at }.from(nil).to(Time)
+
+      expect(event.worker_id).to eq("this_worker")
+    end
+  end
+
+  describe "#process" do
```

Comment:
No there is a convention to use "." for class method and "#" for instance methods. Not sure how wide spread it is. Do we follow a different rule?